legend-engine-xt-relationalStore-store-entitlement-analytics/src/test/java/org/finos/legend/engine/entitlement/TestRelationalStoreEntitlementAnalytics.java

```diff
@@ -0,0 +1,80 @@
+import org.finos.legend.engine.language.pure.compiler.Compiler;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.grammar.from.PureGrammarParser;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.engine.shared.core.deployment.DeploymentMode;
+import org.finos.legend.engine.shared.core.deployment.DeploymentStateAndVersions;
+import org.finos.legend.pure.generated.Root_meta_pure_runtime_Runtime;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URL;
+import java.util.List;
+
+public class TestRelationalStoreEntitlementAnalytics
+{
+    private static final ObjectMapper entitlementObjectMapper = EntitlementModelObjectMapperFactory.getNewObjectMapper();
+
+    private String getResourceAsString(String path)
+    {
+        try
+        {
+            URL infoURL = DeploymentStateAndVersions.class.getClassLoader().getResource(path);
+            if (infoURL != null)
+            {
+                java.util.Scanner scanner = new java.util.Scanner(infoURL.openStream()).useDelimiter("\\A");
+                return scanner.hasNext() ? scanner.next() : null;
+            }
+            return null;
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void testStoreEntitlementAnalyticsRelationalStoreExtension(String modelFilePath, String runtimePath, String mappingPath, String expectedAnalysisResult) throws Exception
+    {
+        String pureModelString = getResourceAsString(modelFilePath);
+        PureModelContextData pureModelContextData = PureGrammarParser.newInstance().parseModel(pureModelString, false);
+        PureModel pureModel = Compiler.compile(pureModelContextData, DeploymentMode.TEST, null);
+        Mapping mapping = pureModel.getMapping(mappingPath);
+        Root_meta_pure_runtime_Runtime runtime = pureModel.getRuntime(runtimePath);
+        RelationalDatabaseEntitlementServiceExtension extension = new RelationalDatabaseEntitlementServiceExtension();
+        List<DatasetSpecification> datasets = extension.generateDatasetSpecifications(null, runtimePath, runtime, mappingPath, mapping, pureModelContextData, pureModel);
+        Assert.assertEquals(entitlementObjectMapper.writeValueAsString(datasets), expectedAnalysisResult);
+    }
+
+    @Test
+    public void testStoreEntitlementAnalyticsForH2() throws Exception
+    {
+        testStoreEntitlementAnalyticsRelationalStoreExtension("models/relationalModel.pure", "runtime::H2Runtime", "mapping::CovidDataMapping", "[{\"name\":\"default.DEMOGRAPHICS\",\"type\":\"H2\",\"database\":\"CovidDataStore\",\"schema\":\"default\",\"table\":\"DEMOGRAPHICS\"},{\"name\":\"default.COVID_DATA\",\"type\":\"H2\",\"database\":\"CovidDataStore\",\"schema\":\"default\",\"table\":\"COVID_DATA\"}]");
```

Comment:
TODO: add a test where you have a runtime that has multiple connection and the mapping refers to a store that might use both connection